lib/DateTime.php

```diff
@@ -705,21 +705,28 @@ public function change(array $options, $cascade=false)
 			'day' => null,
 			'hour' => null,
 			'minute' => null,
-			'second' => null
+			'second' => null,
+			'timezone' => null,
 
 		];
 
-		$options = array_intersect_key($options + $default_options, $default_options);
+		$options = array_intersect_key(array_merge($default_options, $options), $default_options);
```

Comment:
To make the array merge worked expected, using the `array_merge` to merge arrays instead.

When the option values are possible to be override, the `$options` will override the `$default_options` values.

Reply:
I'd rather use unions with default values. Let's revert this one.

Reply:
Hi @olvlvl, thanks for your reply.

Just notice that the union array will have the unexpected behavior when doing following code snippets:

```php
$var = array_merge([4,5,6,7,8], [1,2,3]);
print_r($var) . PHP_EOL;

$var = [4,5,6,7,8] + [1,2,3];
print_r($var) . PHP_EOL;
```

The results are as follows:

```php
Array
(
    [0] => 4
    [1] => 5
    [2] => 6
    [3] => 7
    [4] => 8
    [5] => 1
    [6] => 2
    [7] => 3
)
Array
(
    [0] => 4
    [1] => 5
    [2] => 6
    [3] => 7
    [4] => 8
)
```

But I think it's fine to revert to union array approach because we let the associate arrays do union :).

Reply:
Unions only make sense when key matters, you wouldn't use union on arrays with numeric keys unless these keys really matter. We're using strings for keys here, a union is a perfect candidate.